pkg/languagedetection/client/client.go

```diff
@@ -0,0 +1,282 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package client holds the client to send data to the Cluster-Agent
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	pbgo "github.com/DataDog/datadog-agent/pkg/proto/pbgo/process"
+	"github.com/DataDog/datadog-agent/pkg/util/backoff"
+	"github.com/DataDog/datadog-agent/pkg/util/clusteragent"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+	"github.com/benbjohnson/clock"
+)
+
+const (
+	subscriber          = "language_detection_client"
+	runningMetricPeriod = 1 * time.Minute
+
+	minBackoffFactor = 2.0
+	baseBackoffTime  = 1.0 * time.Second
+	recoveryInterval = 2 * time.Second
+	maxError         = 10
+	maxBackoffTime   = 30 * time.Second
+)
+
+type containerDetails struct {
```

Comment:
I think it would make more sense to export these defined types (i.e use capitalised type name). This can help with code re-usability for development of the handler on the DCA server side. If these types were exported from this package, they can be re-used to parse the request body in the DCA server handler. (same for all struct definitions in this PR)

What do you think?

Reply:
I am not sure we need to export them. I think the protobuf messages should be easily manipulable.